### PR TITLE
Add inital filter to useDataGridRemote hook

### PR DESCRIPTION
**Pull request by Tobi98**
This PR enhances the `useDataGridRemote` hook with an additional `initialFilter` prop.
With this prop inital filter can be set to the dataGrid.

Why? In my projects it is often the case that I initially want to filter by data.

<details>
    <summary>Screenshots/screencasts</summary>

https://github.com/vivid-planet/comet/assets/34998750/9388c71c-792f-4945-8f8e-2e0fae4ae47b

</details>

### CODE REVIEW

**Review by thomasdax98 — DISMISSED** [review]

**Review by nsams — APPROVED** [review]
Thank you for this polished, complete PR.

A possible alternative could be to AND combine the initial filter, support was added to the admin-generator in https://github.com/vivid-planet/comet/pull/1862
(those filters won't show in the UI, but often that's wanted)

File: .changeset/late-tools-design.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+"@comet/admin": minor
+---
+
+Add optional `initalFilter` Prop for useDataGridRemote hook
```
Comment (johnnyomair) [inline .changeset/late-tools-design.md:5]:
```suggestion
Add optional `initalFilter` prop for useDataGridRemote hook
```

File: .changeset/late-tools-design.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+**Example usage:**
+
+```diff
```
Comment (johnnyomair) [inline .changeset/late-tools-design.md:9]:
```suggestion
```tsx
```

No need for diff here

**Review by johnnyomair — APPROVED** [review]

**Comment by Tobi98** [comment]
> Thank you for this polished, complete PR.
> 
> A possible alternative could be to AND combine the initial filter, support was added to the admin-generator in #1862 (those filters won't show in the UI, but often that's wanted)

@nsams yes sounds good for most of the cases 👍🏼  but in the "AND combination" case the filter is always applied to the api query. 
But what if I want to set the filter initially, for example an "active" status, but want to replace it with a different status in the UI without the previous status? Would that be possible with the "AND combination"?

**Comment by nsams** [comment]
> Would that be possible with the "AND combination"?

Of course not, we need both.

File: .changeset/late-tools-design.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+"@comet/admin": minor
+---
+
+Add optional `initalFilter` prop for useDataGridRemote hook
```
Comment (johnnyomair) [inline .changeset/late-tools-design.md:5]:
```suggestion
useDataGridRemote: Add `initialFilter` option
```

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**